packages/passwordless-auth-construct/src/sign-up/construct.ts

```diff
@@ -0,0 +1,72 @@
+import { Construct } from "constructs";
+import { CustomAuthTriggers } from "../types.js";
+import { NodejsFunction, NodejsFunctionProps, OutputFormat } from "aws-cdk-lib/aws-lambda-nodejs";
+import * as path from 'path';
+import { fileURLToPath } from 'url';
+import { Architecture, Runtime } from "aws-cdk-lib/aws-lambda";
+import * as apigateway from "aws-cdk-lib/aws-apigateway";
+import { IUserPool } from "aws-cdk-lib/aws-cognito";
+import { Effect, Policy, PolicyStatement } from "aws-cdk-lib/aws-iam";
+
+const filename = fileURLToPath(import.meta.url);
+const dirname = path.dirname(filename);
+
+export class AmplifySignUpPasswordless extends Construct {
+    constructor(
+        scope: Construct,
+        id: string,
+        triggers: CustomAuthTriggers,
+        userPool: IUserPool
+    ) {
+        super(scope, id);
+
+        // default memory allocation for lambda functions
+        const defaultMemorySize = 128;
+
+        const commonOptions: NodejsFunctionProps = {
+            entry: path.join(dirname, '.', 'createUserService.js'),
+            runtime: Runtime.NODEJS_20_X,
+            architecture: Architecture.ARM_64,
+            bundling: {
+                format: OutputFormat.ESM,
+            },
+        };
```

Comment:
suggestion: we should have these settings defined in one spot if they are going to be similar to https://github.com/aws-amplify/amplify-backend/blob/feat/passwordless-auth/packages/passwordless-auth-construct/src/construct.ts#L46-L53